### PR TITLE
Refactor C# code and switch build to xplat powershell

### PR DESCRIPTION
**Pull request by Redth**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-	"name": "mono-debug",
-	"displayName": "Mono Debug",
-	"version": "0.15.7",
+	"name": "xamarin",
+	"displayName": "Xamarin",
```
Comment (joshspicer) [inline package.json:3]: I should probably flip this back before we merge into microsoft's repo?

File: package.json [hunk 2]
```diff
@@ -14,27 +14,28 @@
 	"license": "MIT",
 	"private": true,
 	"scripts": {
-		"prepublish": "make build",
-		"vscode:prepublish": "make build",
-		"compile": "make build",
+		"prepublish": "powershell ./build.ps1 build",
```
Comment (mcumming) [inline package.json:17]: Shouldn't this be `pwsh` for xplat Powershell Core?
```
mcumming@V-VEDEEP-VM ~ % powershell
zsh: command not found: powershell
mcumming@V-VEDEEP-VM ~ % pwsh
PowerShell 6.2.3
Copyright (c) Microsoft Corporation. All rights reserved.

https://aka.ms/pscore6-docs
Type 'help' to get help.

PS /Users/mcumming> 
```

File: package.json (hunk 1 above)
Comment (Redth) [reply package.json:3]: It raises a good question. We may want to make this it's own extension tbh.

File: package.json (hunk 2 above)
Comment (Redth) [reply package.json]: Thanks fixed 🙂

File: package.json (hunk 1 above)
Comment (joshspicer) [reply package.json:3]: we probably do want to consider this a new extension

**Pull request closed without merging**